### PR TITLE
Dependency clean up

### PR DESCRIPTION
**Pull request by Hyperkid123**
jira: https://issues.redhat.com/browse/RHCLOUD-24470

### Changes
- use swc-loader insetad of babel and ts loaders
- update majority of outdated dependencies
- update some TS errors
- update PF to latest stable version

### CODE REVIEW

File: cypress.config.ts [hunk 1]
```diff
@@ -1,39 +1,38 @@
-import { defineConfig } from "cypress";
-const {
-  addMatchImageSnapshotPlugin,
-} = require('@simonsmith/cypress-image-snapshot/plugin');
+/* eslint-disable @typescript-eslint/no-var-requires */
+import { defineConfig } from 'cypress';
+const { addMatchImageSnapshotPlugin } = require('@hyperkid/cypress-image-snapshot/plugin');
```
Comment (karelhala) [inline cypress.config.ts:3]: Any reason for custom cypress image? I'd be more comfortable with forking it to RHInsights repository and deploying it under our org.

**Review by karelhala — COMMENTED** [review]
Looking good! Just one question about custom cypress image.

File: cypress.config.ts (hunk 1 above)
Comment (Hyperkid123) [reply cypress.config.ts:3]: Yeah, it has locked peer dependency of cypress... I will open a PR in the original package but don't want to wait. I also don't want to take ownership of the package. I see this only as a temporary swap.

Comment (Hyperkid123) [reply cypress.config.ts]: @karelhala fix was published in https://github.com/simonsmith/cypress-image-snapshot/pull/4. I've used the original dependency.

**Review by karelhala — APPROVED** [review]

**Pull request merged**